src/main.rs

```diff
@@ -34,17 +34,30 @@ impl Component for Model {
 
     fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
         match msg {
-            Msg::Update(content) => {
-                match content {
-                    None => (),
-                    Some(content) => self.input = content,
+            Msg::Update(event) => {
+                match event.input_type().as_str() {
+                    "deleteContentBackward" => {
+                        self.input.pop();
+                    }
+                    "insertText" => {
+                        event.data().as_ref().map(|data| self.input.push_str(data));
+                    }
+                    &_ => unimplemented!("\"{}\" is not handled", event.input_type().as_str()),
```

Comment:
I assume there's no way of matching directly on the values returned by `input_type()` instead of converting them to strings?

Reply:
No, I didn't find https://docs.rs/web-sys/latest/web_sys/struct.InputEvent.html#method.input_type

Reply:
The value returned by `input_type()` is a `String`